client/rest-high-level/src/main/java/org/elasticsearch/client/ml/ForecastJobRequest.java

```diff
@@ -47,11 +51,20 @@
             (request, val) -> request.setDuration(TimeValue.parseTimeValue(val, DURATION.getPreferredName())), DURATION);
         PARSER.declareString(
             (request, val) -> request.setExpiresIn(TimeValue.parseTimeValue(val, EXPIRES_IN.getPreferredName())), EXPIRES_IN);
+        PARSER.declareField(ForecastJobRequest::setMaxModelMemory, (p, c) -> {
+            if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
+                return ByteSizeValue.parseBytesSizeValue(p.text(), MAX_MODEL_MEMORY.getPreferredName());
+            } else if (p.currentToken() == XContentParser.Token.VALUE_NUMBER) {
+                return new ByteSizeValue(p.longValue());
+            }
+            throw new IllegalArgumentException("Unsupported token [" + p.currentToken() + "]");
```

Comment:
I know we do it like this in other places, however: There is a `XContentParseException` which I think fits better (its a specialization of IAE).